bc_obps/service/tests/operation_service/test_operation_service_bcghgid.py

```diff
@@ -0,0 +1,146 @@
+
+        assert result.id == manual_bcghg_id
+        assert result.issued_by_id == cas_director.user_guid
+        assert result.comments == 'bcghg id manually set to operation'
+
+        operation.refresh_from_db()
+        assert operation.bcghg_id.id == manual_bcghg_id
+
+    def test_generate_bcghg_id_with_existing_manual_id(self):
+        """Test using an existing manually set BCGHG ID"""
+        cas_director = baker.make_recipe('registration.tests.utils.cas_director')
+        operation = baker.make_recipe('registration.tests.utils.operation', status=Operation.Statuses.REGISTERED)
+        baker.make_recipe('registration.tests.utils.facility', operation=operation)
+
+        existing_bcghg_id = "11234560000"
+
+        # Create existing BCGHG ID
+        BcGreenhouseGasId.objects.create(id=existing_bcghg_id, issued_by_id=cas_director.user_guid, comments='test')
+
+        result = OperationService.generate_bcghg_id(cas_director.user_guid, operation.id, existing_bcghg_id)
+
+        assert result.id == existing_bcghg_id
+
+        operation.refresh_from_db()
+        assert operation.bcghg_id.id == existing_bcghg_id
+
+    def test_generate_bcghg_id_sfo_with_manual_id_updates_facility(self):
+        """Test that SFO facilities get the manually set BCGHG ID"""
+        cas_director = baker.make_recipe('registration.tests.utils.cas_director')
+        operation = baker.make_recipe(
+            'registration.tests.utils.operation', status=Operation.Statuses.REGISTERED, type=Operation.Types.SFO
+        )
+        facility = baker.make_recipe('registration.tests.utils.facility', operation=operation)
+        manual_bcghg_id = "11234567890"
+
+        result = OperationService.generate_bcghg_id(cas_director.user_guid, operation.id, manual_bcghg_id)
+
+        facility.refresh_from_db()
+        operation.refresh_from_db()
+
+        assert operation.bcghg_id.id == manual_bcghg_id
+        assert facility.bcghg_id.id == manual_bcghg_id
+        assert result.id == manual_bcghg_id
+
+    def test_generate_bcghg_id_lfo_with_manual_id_does_not_update_facility(self):
+        """Test that LFO facilities get the manually set BCGHG ID"""
+        cas_director = baker.make_recipe('registration.tests.utils.cas_director')
+        operation = baker.make_recipe(
+            'registration.tests.utils.operation', status=Operation.Statuses.REGISTERED, type=Operation.Types.SFO
+        )
+        facility = baker.make_recipe('registration.tests.utils.facility', operation=operation)
+        manual_bcghg_id = "11234567890"
+
+        result = OperationService.generate_bcghg_id(cas_director.user_guid, operation.id, manual_bcghg_id)
+
+        facility.refresh_from_db()
+        operation.refresh_from_db()
+
+        assert operation.bcghg_id.id == manual_bcghg_id
+        assert facility.bcghg_id.id == manual_bcghg_id
+        assert result.id == manual_bcghg_id
```

Comment:
I'm confused by this test. The name of the test suggests that when an LFO receives a BCGHG ID (applied manually), it should only be applied to the operation and not the facilities within it. This is the behaviour that I would expect. But then the actual logic in the test asserts that the facility receives the same BCGHG ID as the operation?